builder/simplify.cc

```diff
@@ -422,16 +422,16 @@ class simplifier : public node_mutator {
           };
           if (prove_true(crop->bounds.max + 1 >= next_iter.min || next_iter.max + 1 >= crop->bounds.min)) {
             result = crop->body;
-            // If the crop negates the loop variable, the min could become the max. Just do both and take the union.
-            interval_expr new_crop_a = {
-                substitute(crop->bounds.min, op->sym, bounds.min),
-                substitute(crop->bounds.max, op->sym, bounds.max),
-            };
-            interval_expr new_crop_b = {
-                substitute(crop->bounds.min, op->sym, bounds.max),
-                substitute(crop->bounds.max, op->sym, bounds.min),
-            };
-            new_crops.emplace_back(crop->sym, crop->dim, new_crop_a | new_crop_b);
+
+            interval_expr bounds_of_min = bounds_of(crop->bounds.min, {{op->sym, bounds}});
```

Comment:
I think we should add a new helper `interval_expr bounds_of(interval_expr, ...)`, this same pattern occurs in both places you made this change.